htdocs/core/lib/files.lib.php

```diff
@@ -3473,6 +3473,26 @@ function dol_check_secure_access_document($modulepart, $original_file, $entity,
 			$accessallowed = 1;
 		}
 		$original_file = $conf->member->dir_output.'/'.$original_file;
+	} elseif ($modulepart == 'ticket' && !empty($conf->ticket->multidir_output[$entity])) {
+		// Wrapping for events
+		if ($fuser->hasRight('ticket', $read)) {
+			$accessallowed = 1;
+		}
+		$email_split = explode('@', $_SESSION['email_customer']);
+		// include_once DOL_DOCUMENT_ROOT.'/ticket/class/ticket.class.php';
+		$sqlprotectagainstexternals = 'WITH tickets_roles (rowid) AS (SELECT rowid FROM '.MAIN_DB_PREFIX.'c_type_contact WHERE element = "ticket")';
```

Comment:
The use of "WITH" inside a request is very strange and generates compatibility troubles. In most cases, a common SQL can work.
Here we want a SQL to test if a record is correctly linked to a some data with a filter on an email. A simple SELECT WHERE EXISTS (SELECT ...) should do the job with a better execution plan for the database.
Here for exemple you can use replace
`LEFT JOIN tickets_roles tr ON tr.rowid = ec.fk_c_type_contact`
by
`WHERE tr.rowid IN (SELECT rowid FROM '.MAIN_DB_PREFIX.'c_type_contact WHERE element = "ticket")`

Reply:
I chenged this in #32395.